### PR TITLE
Add toString() method to Ballerina Objects

### PR DESCRIPTION
**Pull request by gimantha**
## Purpose
This PR adds toString() method to ballerina objects. For more details about the behavior of toString on object, refer to [https://raw.githubusercontent.com/ballerina-platform/ballerina-spec/v2019R2/lang/lib/value.bal](url)

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [x] Added necessary documentation  
  - [x] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ArrayValue.java [hunk 1]
```diff
@@ -495,30 +499,7 @@ private int getCurrentArrayLength() {
...
-            } else if (elementType.getTag() == TypeTags.FLOAT_TAG) {
-                for (int i = 0; i < size; i++) {
-                    sj.add(Double.toString(floatValues[i]));
-                }
-                return sj.toString();
-            } else if (elementType.getTag() == TypeTags.STRING_TAG) {
-                for (int i = 0; i < size; i++) {
-                    sj.add(stringValues[i]);
-                }
+            if (generateStringForBasicTypes(sj)) {
```
Comment (hasithaa) [inline bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ArrayValue.java]: Shall we refactor this, the method looks like validation, but it does more than that.

Comment (gimantha) [reply bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ArrayValue.java]: will refactor it

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/utils/StringUtils.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import org.ballerinalang.jvm.types.BType;
+import org.ballerinalang.jvm.types.TypeTags;
+import org.ballerinalang.jvm.values.AbstractObjectValue;
+import org.ballerinalang.jvm.values.ArrayValue;
+import org.ballerinalang.jvm.values.MapValueImpl;
+import org.ballerinalang.jvm.values.RefValue;
+
+/**
+ * This class contains the methods that deals with Strings.
+ */
```
Comment (hasithaa) [inline bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/utils/StringUtils.java]: Since is missing.

**Comment by hasithaa** [comment]
```ballerina
import ballerina/io;
public function main() {
    map<string> yy = {a :"aa", b: "bb"};
    io:println(yy.toJsonString());
}
```

Above doesn't give the expected output.

**Comment by gimantha** [comment]
> ```ballerina
> import ballerina/io;
> public function main() {
>     map<string> yy = {a :"aa", b: "bb"};
>     io:println(yy.toJsonString());
> }
> ```
> 
> Above doesn't give the expected output.

@hasithaa This should fail right? map<string> is not a json and toJsonString takes a json as its argument

**Comment by gimantha** [comment]
> ```ballerina
> import ballerina/io;
> public function main() {
>     map<string> yy = {a :"aa", b: "bb"};
>     io:println(yy.toJsonString());
> }
> ```
> 
> Above doesn't give the expected output.

PR sent to fix this. #17300

**Pull request closed without merging**